street-explorer/www/js/lane_editor.js

```diff
@@ -0,0 +1,205 @@
+    const url = `https://overpass-api.de/api/interpreter?data=${query}`;
+    console.log(`Fetching from overpass: ${url}`);
+
+    importButton.innerText = "Downloading from Overpass...";
+    // Prevent this function from happening twice in a row. It could also maybe
+    // be nice to allow cancellation.
+    importButton.disabled = true;
+
+    try {
+      const resp = await fetch(url);
+      const osmXML = await resp.text();
+
+      importButton.innerText = "Importing OSM data...";
+
+      this.network = new JsStreetNetwork(osmXML, {
+        debug_each_step: false,
+        dual_carriageway_experiment: false,
+        cycletrack_snapping_experiment: false,
+        inferred_sidewalks: false,
+        osm2lanes: false,
+      });
+      this.rerenderAll();
+      const bounds = this.layers[0].getBounds();
+      this.map.fitBounds(bounds, { animate: false });
+    } catch (err) {
+      window.alert(`Import failed: ${err}`);
+    } finally {
+      // Make the button clickable again
+      importButton.innerText = "Import current view";
+      importButton.disabled = false;
+    }
+  }
+
+  rerenderAll() {
+    for (const layer of this.layers) {
+      layer.remove();
+    }
+    this.layers = [];
+
+    this.layers.push(
+      L.geoJSON(JSON.parse(this.network.toGeojsonPlain()), {
+        // Just show intersections from the plain layer
+        style: (feature) => {
+          if (feature.properties.type == "intersection") {
+            return {
+              color: "black",
+              fillOpacity: 0.7,
+            };
+          }
+          return { fill: false, stroke: false };
+        },
+      })
+    );
+    this.layers.push(
+      L.geoJSON(JSON.parse(this.network.toLanePolygonsGeojson()), {
+        style: lanePolygonStyle,
+        // Make lanes clickable
+        onEachFeature: (feature, layer) => {
+          layer.on({
+            click: (ev) => {
+              if (feature.properties.osm_way_ids.length != 1) {
+                window.alert(
+                  "This road doesn't match up with one OSM way; you can't edit it"
+                );
+              } else {
+                this.editWay(BigInt(feature.properties.osm_way_ids[0]));
+              }
+            },
+          });
+        },
+      })
+    );
+    this.layers.push(
+      makeLaneMarkingsLayer(this.network.toLaneMarkingsGeojson())
+    );
+
+    for (const layer of this.layers) {
+      layer.addTo(this.map);
+    }
+  }
+
+  editWay(id) {
+    this.currentWay = id;
+    if (this.currentWaysLayer) {
+      this.currentWaysLayer.remove();
+    }
+    this.currentWaysLayer = L.geoJSON(
+      JSON.parse(this.network.getGeometryForWay(id)),
+      {
+        style: (feature) => {
+          return { stroke: false, fill: true, color: "red", opacity: 0.5 };
+        },
+      }
+    ).addTo(this.map);
+
+    var html = `<table><tbody id="tags-table">`;
+
+    const tags = JSON.parse(this.network.getOsmTagsForWay(id));
```

Comment:
There's a mix of setting `innerHTML` and using DOM methods here. It's a bit messy, but seems correct.

Maybe if iD ever has a standalone library for something like the tags table component, it'd be cool to automatically share support for stuff like autocomplete for tag values.